src/nlp/utils.py

```diff
@@ -0,0 +1,20 @@
+def preprocess_title(sentence: str) -> str:
+    """Method to preprocess title:
+        - lowercase the title
+        - remove trailing spaces
+        - remove encoding errors
+        - remove punctuation
+        - remove metrics (cm, mm, ml etc)
+
+    Args:
+        sentence (str): title string
+
+    Returns:
+        str: preprocessed title
+    """
+    sentence = sentence.lower()
+    sentence = sentence.replace(r"[^\w\s]", "")
+    sentence = sentence.replace(r"\s+$", "")
+    sentence = sentence.replace(r"x\d+$", "")
+    sentence = sentence.replace(r"\W*\b\w{1,3}\b", "")
```

Comment:
it doesn't remove any punctuation from a sentence

Reply:
Fixed, str.replace doesn't allow to use regex, used re.sub instead in utils.py